src/bp-core/bp-core-functions.php

```diff
@@ -4332,49 +4332,58 @@ function bp_core_get_active_custom_post_type_feed() {
  */
 function bp_platform_default_activity_types() {
 
-	$activity_type = apply_filters(
-		'bp_platform_default_activity_types',
+	$settings_fields = array(
 		array(
-			'0' => array(
-				'activity_name'  => 'new_avatar',
-				'activity_label' => __( 'Member changes their profile photo', 'buddyboss' ),
-			),
-			'1' => array(
-				'activity_name'  => 'updated_profile',
-				'activity_label' => __( 'Member updates their profile details', 'buddyboss' ),
-			),
-			'2' => array(
-				'activity_name'  => 'new_member',
-				'activity_label' => __( 'Member registers to the site', 'buddyboss' ),
-			),
-			'3' => array(
-				'activity_name'  => 'friendship_created',
-				'activity_label' => __( 'Two members become connected', 'buddyboss' ),
-			),
-			'4' => array(
+			'activity_name'  => 'new_avatar',
+			'activity_label' => __( 'Member changes their profile photo', 'buddyboss' ),
+		),
+		array(
+			'activity_name'  => 'updated_profile',
+			'activity_label' => __( 'Member updates their profile details', 'buddyboss' ),
+		),
+		array(
+			'activity_name'  => 'new_member',
+			'activity_label' => __( 'Member registers to the site', 'buddyboss' ),
+		),
+		array(
+			'activity_name'  => 'friendship_created',
+			'activity_label' => __( 'Two members become connected', 'buddyboss' ),
+		),
+	);
+
+	// settings field that dependend on group
+	if ( bp_is_active( 'groups' ) ) {
+		$settings_fields = array_merge( $settings_fields, array(
+			array(
 				'activity_name'  => 'created_group',
 				'activity_label' => __( 'Member creates a group', 'buddyboss' ),
 			),
-			'5' => array(
+			array(
 				'activity_name'  => 'joined_group',
 				'activity_label' => __( 'Member joins a group', 'buddyboss' ),
 			),
-			'6' => array(
+			array(
 				'activity_name'  => 'group_details_updated',
 				'activity_label' => __( 'Group details are updated', 'buddyboss' ),
 			),
-			'7' => array(
+		) );
+	}
+
+	// Settings field that dependent on forum
+	if ( bp_is_active( 'forums' ) ) {
+		$settings_fields = array_merge( $settings_fields, array(
+			array(
 				'activity_name'  => 'bbp_topic_create',
 				'activity_label' => __( 'Member creates a forum discussion', 'buddyboss' ),
 			),
-			'8' => array(
+			array(
 				'activity_name'  => 'bbp_reply_create',
 				'activity_label' => __( 'Member replies to a forum discussion', 'buddyboss' ),
 			),
-		)
-	);
+		) );
+	}
 
-	return $activity_type;
+	return apply_filters( 'bp_platform_default_activity_types', $settings_fields );
```

Comment:
all new filters should start with `bb_` instead of `bp_`

also missed the doc block of the filter with proper info.